src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs

```diff
@@ -0,0 +1,424 @@
+            where T : OpenXmlPart, ISupportRelationship<EmbeddedControlPersistencePart>
+        {
+            EmbeddedControlPersistencePart childPart = new();
+
+            if (parent == null)
+            {
+                throw new ArgumentNullException(nameof(parent));
+            }
+
+            parent.InitPart(childPart, contentType, id);
+            return childPart;
+        }
+
+        /// <summary>
+        /// Adds a EmbeddedObjectPart as a relationship to the parent part
+        /// </summary>
+        /// <param name="parent">The parent part requesting to add.</param>
+        /// <param name="partType">The part type information for the added extensible part.</param>
+        /// <param name="id">The relationship id. Optional, default to null.</param>
+        /// <return>The newly added part</return>
+        /// <see cref="EmbeddedObjectPart"/>
+        public static EmbeddedObjectPart AddEmbeddedObjectPart<T>(this T parent, PartTypeInfo partType, string? id = null)
+            where T : OpenXmlPart, ISupportRelationship<EmbeddedObjectPart> => OpenXmlPartExtensions.InitPart<EmbeddedObjectPart>(parent, new EmbeddedObjectPart(), partType, id);
+
+        /// <summary>
+        /// Adds a EmbeddedObjectPart as a relationship to the parent part
+        /// </summary>
+        /// <param name="parent">The parent part requesting to add.</param>
+        /// <param name="contentType">The content type information for the added extensible part.</param>
+        /// <param name="id">The relationship id. Optional, default to null.</param>
+        /// <return>The newly added part</return>
+        /// <see cref="EmbeddedObjectPart"/>
+        public static EmbeddedObjectPart AddEmbeddedObjectPart<T>(this T parent, string contentType, string? id = null)
+            where T : OpenXmlPart, ISupportRelationship<EmbeddedObjectPart>
+        {
+            EmbeddedObjectPart childPart = new();
+
+            if (parent == null)
+            {
+                throw new ArgumentNullException(nameof(parent));
+            }
+
+            parent.InitPart(childPart, contentType, id);
+            return childPart;
+        }
+
+        /// <summary>
+        /// Adds a EmbeddedPackagePart as a relationship to the parent part
+        /// </summary>
+        /// <param name="parent">The parent part requesting to add.</param>
+        /// <param name="partType">The part type information for the added extensible part.</param>
+        /// <param name="id">The relationship id. Optional, default to null.</param>
+        /// <return>The newly added part</return>
+        /// <see cref="EmbeddedPackagePart"/>
+        public static EmbeddedPackagePart AddEmbeddedPackagePart<T>(this T parent, PartTypeInfo partType, string? id = null)
+            where T : OpenXmlPart, ISupportRelationship<EmbeddedPackagePart> => OpenXmlPartExtensions.InitPart<EmbeddedPackagePart>(parent, new EmbeddedPackagePart(), partType, id);
```

Comment:
```suggestion
            where T : OpenXmlPart, ISupportRelationship<EmbeddedPackagePart>
            => parent.InitPart(new EmbeddedPackagePart(), partType, id);
```

and so on for the others

Reply:
that's the way I had it initially. ok, I'll change them all back. :/